### ISSUE TITLE
tf.keras.models.load_model unable to use output layer names specified for loss_weights

### ISSUE THREAD

**Issue by SinghB** [tensorflow/tensorflow#35665]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 64-Bit
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v2 via conda install tensorflow-gpu cudnn=7.3.1=cuda9.0_0
- Python version: 3.7.4
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: cudnn=7.3.1=cuda9.0_0
- GPU model and memory: NVIDIA Quadro M1200

**Describe the current behavior**
I am using tf.keras functional APIs to create a multi-output model and to my compile call, I provide loss weights specific to my output names:

```
my_model.compile(optimizer=optimizer, metrics=['accuracy']
                      ,loss='categorical_crossentropy'
                      ,loss_weights={'sec': 0.25, 'cls': 0.35, 'subcls': 0.4}
                     )
```

Model compilation, fitment all work and I'm able to use model object to predict, evaluate etc. I'm also able to save model in 'tf' format. The save for 'hdf5' format doesn't work and gives OOM error, but that's ok for me now. The code to save is:

`model.save(r'./models/hf_s100/')`

The problem I face is when I try to load it via:

`hf_model = keras.models.load_model(r'./models/hf_s100/')`

I get an error that loss weights found to be 'sec', 'cls' and 'subcls' and was expecting 'output_1', 'output_2', 'output_3'

**Describe the expected behavior**
Given that we are allowed to name our output layers and use the names in loss_weights option, the load_model should also be able to pick the name specified as part of model save.

**Code to reproduce the issue**
See code snippets provided above

**Comment by ravikyram** [comment]
@SinghB 

Can you please share simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ravikyram** [comment]
@SinghB 

Any update on this issue please. Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
